### PR TITLE
[MiHome] Add a Bridge property for the network interface to be used for multicast traffic

### PR DESCRIPTION
**Pull request by ratza**
[MiHome] Add a Bridge property for the network interface to be used for multicast traffic

This MR adds a property on the Bridge Thing to allow explicit configuration of the network interface name to which the MulticastSocket is bound.

When the computer running OpenHAB has more than one network interface configured (typically, a VLAN for your segregated IoT devices, and the other for your regular traffic like internet, OpenHAB panel access, etc), it could be that OpenHAB will attempt to listen for Multicast traffic of the Gateway on the wrong network interface. Depending on the OS the network interface chosen for binding the Multicast traffic depends on routing rules, default gateway or plain interface order (don't quote me on that). Couldn't find a specific document that describes the actual behavior, mostly because multiple networks configured for multicast is not considered the norm, hence has an undefined behavior. The solution is to bind to a specific network interface. Since there's no option to do it globally (for all Bindings) within OpenHAB and I read, somewhere on the forum, that it is expected for each binding to implement support separately, I've added this for the MiHome binding. I've also added a succinct description to the documentation. 

Provides a graceful property to solve the issue described in #3239.

I've tested this in my own environment (Ubuntu, 2 VLANs), to be working correctly.

Signed-off-by: Ion Marusic <ion.marusic@gmail.com>

### CODE REVIEW

**Comment by fwolter** [comment]
@pboos can you take a look?

File: bundles/org.openhab.binding.mihome/README.md [hunk 1]
```diff
@@ -110,13 +110,17 @@ Just follow the instructions in ["Connecting devices to the gateway"](#connectin
 
 - The binding requires port `9898` to not be used by any other service on the system.
 - Make sure multicast traffic is correctly routed between the gateway and your openHAB instance
+- To correctly receive multicast traffic, when your OpenHAB machine is using multiple network interfaces, you might need to configure the optional `interface` property on the `Bridge` Thing, like so:
```
Comment (fwolter) [inline bundles/org.openhab.binding.mihome/README.md:113]: Please check all.
```suggestion
- To correctly receive multicast traffic, when your openHAB machine is using multiple network interfaces, you might need to configure the optional `interface` property on the `Bridge` Thing, like so:
```

File: bundles/org.openhab.binding.mihome/src/main/resources/OH-INF/config/config.xml [hunk 2]
```diff
@@ -36,6 +36,14 @@
 			<advanced>true</advanced>
 		</parameter>
 
+		<parameter name="interface" type="text">
+			<context>network-address</context>
```
Comment (fwolter) [inline bundles/org.openhab.binding.mihome/src/main/resources/OH-INF/config/config.xml:40]: The `network-address` context validates hostnames and IP adresses. AFAIK, there is no context for interface names. So, this should be removed.

**Review by fwolter — CHANGES_REQUESTED** [review]

**Comment by fwolter** [comment]
@ratza What's the state of this PR?

File: bundles/org.openhab.binding.mihome/src/main/resources/OH-INF/config/config.xml (hunk 2 above)
Comment (ratza) [reply bundles/org.openhab.binding.mihome/src/main/resources/OH-INF/config/config.xml]: Removed. Unfortunately, I'm not accustomed with the effect of the "contexts", so I blindly replicated based on the other fields.

File: bundles/org.openhab.binding.mihome/README.md (hunk 1 above)
Comment (ratza) [reply bundles/org.openhab.binding.mihome/README.md]: Reviewed the whole file and updated all relevant instances. Also fixed some other obvious typos.

**Comment by ratza** [comment]
> @ratza What's the state of this PR?

My apologies, life took over. Just pushed the relevant changes.

**Review by fwolter — APPROVED** [review]
Thanks! LGTM

**Pull request merged**